src/build123d/topology.py

```diff
@@ -8592,7 +8379,58 @@ def chamfer_2d(
         Returns:
             Wire: chamfered wire
         """
-        return Face(self).chamfer_2d(distance, distance2, vertices, edge).outer_wire()
+        reference_edge = edge
+        del edge
+
+        # Create a face to chamfer
+        unchamfered_face = _make_topods_face_from_wires(self.wrapped)
+        chamfer_builder = BRepFilletAPI_MakeFillet2d(unchamfered_face)
+
+        vertex_edge_map = TopTools_IndexedDataMapOfShapeListOfShape()
+        TopExp.MapShapesAndAncestors_s(
+            unchamfered_face, ta.TopAbs_VERTEX, ta.TopAbs_EDGE, vertex_edge_map
+        )
+
+        for v in vertices:
+            edges = vertex_edge_map.FindFromKey(v.wrapped)
+
+            # Index or iterator access to OCP.TopTools.TopTools_ListOfShape is slow on M1 macs
+            # Using First() and Last() to omit
+            edges = [edges.First(), edges.Last()]
+
+            # Need to wrap in b3d objects for comparison to work
+            # ref.wrapped != edge.wrapped but ref == edge
+            edges = [Edge(e) for e in edges]
```

Comment:
```suggestion
            # Index or iterator access to OCP.TopTools.TopTools_ListOfShape is slow on M1 macs
            # Using First() and Last() to omit
            # Need to wrap in b3d objects for comparison to work
            # ref.wrapped != edge.wrapped but ref == edge
            edges = Edge(edges.First()), Edge(edges.Last())
```
there's the same bit of code in `chamfer_2d` too

Reply:
Done